### PR TITLE
feat(eval,cli): print why a gated metric failed, not just that it did

### PR DESCRIPTION
**Pull request by kwakayama**
A failing metric line said *that* something failed, never *why*:

```
  ● LLM as a judge passed: 0/1 passed (0%)
```

Everything needed to act on that already existed in `summary.json`, unprinted:

> The answer correctly states the $15.21 tip and $99.71 total, but it does not split the total into $33.24, $33.24, and $33.23; it incorrectly claims $33.24 each.

For a judge rubric the verdict *is* the reason the eval failed. Reading it should not mean opening a JSON artifact.

## After

```
  Eval:   Assistant smoke test
  Target: agent:assistant
  Result: 0/1 passed (0%)

  ● Agent called tool "calculator": 1/1 passed (100%)
  ● Agent had no failed tool calls: 1/1 passed (100%)
  ● LLM as a judge passed: 0/1 passed (0%)
      calculator: The answer correctly states the $15.21 tip and $99.71 total, but ...
```

Explanations print under the metric that produced them, three per metric, then `and N more, see the report`. The cap keeps a large dataset from burying the summary; the pointer says where the rest live.

## Which record errors are suppressed

`summary.gateFailures` also carries a `record.error` entry whenever a record did not complete. Printing it unconditionally would restate the judge verdict directly above it as a second, independent problem.

The rule keys on the *explanation*, not the severity:

```ts
if (
  failure.name === "record.error" &&
  failure.explanation === RECORD_INCOMPLETE_EXPLANATION &&
  derived.has(failure.recordId)
) continue;
```

- `isBlockingFailure` (`src/eval/runner.ts:232`) clears `completed` for **both** `gate` and `budget` severities, so either one yields the synthesized `Record did not complete.` stand-in. Both are suppressed when another failure already explains the record.
- A `record.error` carrying text of its own — an adapter that threw, say — **always** prints. It is the only line explaining why the agent never answered, and nothing else reports it.

The stand-in string was written out at both sites that build it (`report.ts`, `run-report.ts`). It is now an exported `RECORD_INCOMPLETE_EXPLANATION`, so the suppression rule cannot drift from the text it matches.

## Review follow-ups

- **Bounded suppression to the stand-in explanation** ([#discussion_r3736104367](https://github.com/veryfront/veryfront-code/pull/3458#discussion_r3736104367)). CodeRabbit proposed filtering `derived` to `severity === "gate"`. I did not take that: budget failures clear `completed` too, so it would have reintroduced the duplicate line for every budget metric. It did surface a real bug next door — a thrown adapter error being swallowed — which is what the explanation check fixes.
- **Env restore in tests** ([#discussion_r3736104363](https://github.com/veryfront/veryfront-code/pull/3458#discussion_r3736104363)) — skipped with reasoning. Ten tests in this file share the pattern; changing one would make the file inconsistent. Offered as a separate cleanup.

## Testing

- `src/eval/`, `cli/commands/eval/`, `src/extensions/eval/`, `extensions/ext-eval-report-mlflow/`: 21 passed, 255 steps, 0 failed
- `deno lint`, `deno fmt --check` (58 files): clean
- `deno check` on every touched file: clean

Both new tests were verified failing against the previous code first:

- `prints why a gated metric failed, without restating it as a record error`
- `keeps a record error that carries detail of its own` — drives a rejecting adapter plus a failing gate, asserts the upstream message survives

Not from this branch: `deno check src/eval/judges.test.ts` reports three `TS18048 'call' is possibly 'undefined'` errors. They reproduce unchanged on `origin/main`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added explanations for gated evaluation metric failures in CLI output.
  * Grouped failure reasons by metric and summarized additional reasons when more than three apply.
  * Preserved independent upstream errors alongside gate failures.

* **Bug Fixes**
  * Removed redundant incomplete-record messages when a more specific metric explanation is available.
  * Standardized incomplete-record explanations across evaluation and report outputs.

* **Documentation**
  * Updated evaluation API reference links and documented the shared incomplete-record explanation.

### CODE REVIEW

File: cli/commands/eval/command.ts [hunk 1]
```diff
@@ -834,6 +835,56 @@ function formatPercent(rate: number): string {
...
+ * adapter error, keeps its reason because nothing else reports it.
+ */
+function groupGateFailureReasons(
+  failures: EvalGateFailureSummary[],
+): { byMetric: Map<string, string[]>; unattached: string[] } {
+  const derived = new Set(
+    failures.filter((failure) => failure.name !== "record.error").map((failure) =>
+      failure.recordId
+    ),
+  );
```
[the remark this answers is not in the document]
Comment (kwakayama) [reply cli/commands/eval/command.ts:862]: Real bug here, but not the one described, and I did not take the suggested diff — it would make things worse. Fixed differently in 57112cf29.

**Why `severity === "gate"` is the wrong filter.** The premise is that a budget failure leaves `record.error` carrying independent detail. It does not. `isBlockingFailure` covers *both* severities:

```ts
// src/eval/runner.ts:232
function isBlockingFailure(record: EvalRecord): boolean {
  return [...(record.metrics ?? []), ...(record.checks ?? [])].some((result) =>
    !result.skipped && result.pass === false &&
    (result.severity === "gate" || result.severity === "budget")
  );
}
```

and that is what clears `completed` (line 575). So a budget failure produces exactly the same synthesized stand-in as a gate failure:

```ts
// src/eval/report.ts:188
explanation: record.error ?? "Record did not complete.",
```

Restricting `derived` to `gate` would reintroduce the duplicate `Record did not complete.` line for every budget-severity metric — the precise thing this PR removes.

**The real bug it led me to.** The information loss is not about severity at all, it is about whether `record.error` holds real text. When an adapter throws *and* a metric fails, my filter dropped the adapter message — the only line explaining why the agent never answered. `severity === "gate"` does not fix that case either, since the failing metric is usually a gate.

So the condition keys on the explanation rather than the severity:

```ts
if (
  failure.name === "record.error" &&
  failure.explanation === RECORD_INCOMPLETE_EXPLANATION &&
  derived.has(failure.recordId)
) continue;
```

Suppress the stand-in, always keep text of its own. `derived` deliberately still spans both severities.

The string was duplicated across `report.ts` and `run-report.ts`, so it is now an exported `RECORD_INCOMPLETE_EXPLANATION` and the rule cannot drift from the text it matches.

Regression test `keeps a record error that carries detail of its own` drives a rejecting adapter plus a failing gate and asserts the upstream message survives. Verified failing against the previous code first.

File: cli/commands/eval/command.test.ts [hunk 2]
```diff
@@ -1231,6 +1231,79 @@ describe("eval CLI command helpers", () => {
...
+        ),
+        true,
+      );
+      // The runner flips `completed` off whenever a gate fails, so "Record did not complete."
+      // would only restate the judge verdict above it.
+      assertEquals(printed.some((line) => line.includes("Record did not complete")), false);
+    } finally {
+      await Deno.remove(projectDir, { recursive: true });
+      await Deno.remove(configHome, { recursive: true });
+    }
```
[the remark this answers is not in the document]
Comment (kwakayama) [reply cli/commands/eval/command.test.ts:1304]: Skipping this one, with a reason rather than a disagreement.

The concern is fair in the abstract, but the pattern is file-wide, not something this test introduces. Ten tests in `command.test.ts` do the identical thing — delete the four `VERYFRONT_*` vars, point `XDG_CONFIG_HOME` at a fresh temp dir, restore none of it. The `prints no report output under --quiet` test directly below this one is the same shape. Fixing only the new test would leave nine unchanged and make the file inconsistent about a convention it currently applies uniformly.

Leakage is also bounded in practice: every test here sets `XDG_CONFIG_HOME` itself before running, so none of them depends on inheriting a valid one, and the credential vars are deleted rather than set to a stale value.

Happy to do the save-and-restore sweep across all ten as its own change if you want it — it is a test-hygiene cleanup with no behaviour change, and it does not belong in a PR about printing failure reasons.

File: cli/commands/eval/command.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (kwakayama) [reply cli/commands/eval/command.ts:862]: One correction to my reply above, since it is now stale: `RECORD_INCOMPLETE_EXPLANATION` is **not** an exported part of the public API.

I originally re-exported it from the `veryfront/eval` barrel, which turned an internal CLI detail into a supported public export and made `docs/api-reference/veryfront/eval.md` stale — `ci (lint)` caught it. Adding a permanent public export to fix an internal suppression rule was the wrong trade, so the CLI now imports it directly:

```ts
import { RECORD_INCOMPLETE_EXPLANATION } from "../../../src/eval/report.ts";
```

which matches how other CLI modules reach internal `src/` code (`cli/commands/workflow/command.ts`, `cli/commands/init/catalog.ts`). The constant is still shared between `report.ts` and `run-report.ts`, so the anti-drift property you noted holds — it just no longer widens the published surface. The generated API reference is unchanged from `main` apart from two line-number shifts.

Branch is also rebased onto current `main` (9acfc01a7).
